File: Code Generation/responses/python/deepseek/easy/maximum-unique-subarray-sum-after-deletion.py
from typing import List

class Solution:
    def maxSum(self, nums: List[int]) -> int:
        n = len(nums)
        max_sum = float('-inf')
        
        # Try every possible subarray
        for i in range(n):
            seen = set()
            current_sum = 0
            
            for j in range(i, n):
                if nums[j] in seen:
                    break
                seen.add(nums[j])
                current_sum += nums[j]
                max_sum = max(max_sum, current_sum)
        
        return max_sum

File: Code Generation/responses/cpp/deepseek/easy/maximum-unique-subarray-sum-after-deletion.cpp
#include <vector>
#include <unordered_set>
#include <algorithm>
#include <climits>

using namespace std;

class Solution {
public:
    int maxSum(vector<int> nums) {
        int n = nums.size();
        int maxSum = INT_MIN;
        
        // Try all possible starting points of subarray
        for (int i = 0; i < n; i++) {
            unordered_set<int> seen;
            int currentSum = 0;
            
            // Extend the subarray from i to j
            for (int j = i; j < n; j++) {
                // If we encounter a duplicate, break this subarray
                if (seen.find(nums[j]) != seen.end()) {
                    break;
                }
                
                seen.insert(nums[j]);
                currentSum += nums[j];
                maxSum = max(maxSum, currentSum);
            }
        }
        
        return maxSum;
    }
};